### ISSUE TITLE
Unexpected number of vertices when loading from OBJ

### ISSUE THREAD

**Issue by kaufManu** [mikedh/trimesh#489]
First, thanks a lot for this great package, which is very handy!

I'm currently running into some inconsistencies when loading meshes from obj files. Specifically, I noticed that sometimes the number of vertices is higher than what I would expect. For example, considering the three models in the [pyrender example folder](https://github.com/mmatl/pyrender/tree/master/examples) and the following code snippet :

```python
fuze_trimesh = trimesh.load('./models/fuze.obj')
print('fuze mesh # vertices', len(fuze_trimesh.vertices))

drill_trimesh = trimesh.load('./models/drill.obj')
print('drill mesh # vertices', len(drill_trimesh.vertices))

wood_trimesh = trimesh.load('./models/wood.obj')
print('wood mesh # vertices', len(wood_trimesh.vertices))
```

This prints

```
fuze mesh # vertices 664
drill mesh # vertices 34343
wood mesh # vertices 4
```

Except for the wood mesh, the number of vertices is higher then what I expected by inspecting the obj file. The fuze mesh has 502 `v` entries and the drill mesh has 32'770 `v`'s. This is confirmed when I open these obj files in Meshlab, where the vertex count matches the number of `v` entries in the respective file.

**Comment by mikedh** [comment]
Hey, yeah this is probably a result of split indices in an OBJ file and how we represent meshes. For example in the faces:
```
f 1/1/1 2/2/2 3/3/3
f 1/2/3 2/2/2 4/4/4
```

The first vertex has the same position, but different texture and normal.  In the simple mesh representation we use in`trimesh` meshes have `(n, 3)` vertices which have corresponding `(n, 3)` properties (like UV coordinates). 

In the case of meshes like this example, we can't represent it as simple corresponding arrays unless we separate the first vertex into multiple "vertices." Basically if we don't do that, the texture looks like we ran it through a woodchipper. 

Heads up, the OBJ code is being fully re-written (in the major release next week). It will still do this however haha. The relevant code (in the new version) for this split logic is [here](https://github.com/mikedh/trimesh/blob/7ec64a538acdd1475f7ff0e2367c763325832914/trimesh/visual/texture.py#L175-L232). 

If you have suggestions for a better way to handle this, I'm certainly open to them!

**Comment by kaufManu** [comment]
That makes sense now, thanks. Basically I'm just looking for a way to create a new `trimesh` instance from `numpy` vertices and faces (which works well) and then add a texture to it that I load from a different obj file. This obj file encodes exactly the same mesh topology, but the vertex positions might be different.  To make it a bit clearer, what I'm trying to do is:

```python
vertices, faces = ... # some meaningful numpy arrays of shape (n, 3)
mesh = trimesh.Trimesh(vertices=vertices, faces=faces)
textured_mesh = trimesh.load('./texture_mesh.obj', process=False)
mesh.visual = textured_mesh.visual  # this does not end well
```

The texture on `mesh` should now match the texture from `textured_mesh`, but this does not work, because the vertices in `textured_mesh` are different from `mesh` because of what you explained. So, is there a way to replicate the copying of vertices when loading from an obj file for the `mesh` instance?

**Comment by paulmelnikow** [comment]
Side note:

> The first vertex has the same position, but different texture and normal. In the simple mesh representation we use in`trimesh` meshes have `(n, 3)` vertices which have corresponding `(n, 3)` properties (like UV coordinates).

Ahh! Now I think I understand what you've been saying about OBJs and indexing @mikedh. 😀

**Comment by mikedh** [comment]
Closing for now as new OBJ loader is merged (#436) and presumably has different, exciting problems. Feel free to re-open!

**Comment by kaufManu** [comment]
I'm running in this issue again and I'm wondering how to work around it. I understand why the vertices have to be duplicated and that makes perfect sense. However, I'm not sure whether there's a problem with the vertex normals for the vertices that are duplicated. In my application, I'm loading a mesh of a sphere from an OBJ file that includes texture coordinates. When I render this sphere (not showing the texture here) including the vertex normals, it looks like this:

![image](https://user-images.githubusercontent.com/5639197/118107583-3862a580-b3df-11eb-950a-e983aa83fe57.png)

Here you can clearly see the seams of the texture, because the vertices along the seams have multiple vertex normals and hence we see some differences in shading. It makes somewhat sense that this happens - the duplicated vertices belong to different faces, hence when computing the vertex normals based on the face normals of the adjacent faces, we get different results each time. However, the vertex normals are already supplied in the OBJ file that I'm using, so unless trimesh is re-triggering a computation of the vertex normals, I don't understand why this should happen. If I remove the texture coordinates from my OBJ file, then this problem does not appear. So I'm assuming what happens is: 1) trimesh detects that vertices must be duplicated because of the uv-coordinates 2) instead of duplicating the vertex normals, it re-computes them leading to the above artifact.

This assessment might be totally wrong, but it would be great if you could double-check that there is indeed no bug with the vertex normals. If there isn't, then I wonder how I should work around it, i.e. is there a way to get the "un-duplicated" data from trimesh so that I can do the duplication myself?

Many thanks for your help!
